File: TP/TP2/Code/1C.cpp
/* Calculer la somme des N premières puissances de 2 */
#include <cmath> // Contient la fonction pow()
#include <iostream>

/* Fonction */

int somme(int a)
{
  int sum;

  sum = 0;

  for (int i = 0; i < a; ++i)
    sum += pow(2, i);

  return sum;
}

/* Utilisation */

int main()
{
  int n;

  std::cout << "Donne moi une valeur : " << std::flush;
  std::cin >> n;

  std::cout << "La somme de " << n << " premiers puissance de 2 est " 
            << somme(n) << std::endl;

  return 0;
}

File: TP/TP2/Code/Python/1C.py
#!/usr/bin/python
# Calculer la somme des N premières puissances de 2
from math import pow


def sum_pow(val):
    sum_val = 0

    for i in range(val):
        sum_val += pow(2, i)  # I ❤ Python : sum_val += 2 ** i

    return sum_val


print('Donne moi une valeur : ', end='')
foo = int(input())

print('La somme de', foo, 'première puissance de 2 est', sum_pow(foo))
